src/com/owncloud/android/services/observer/InstantUploadFolderObserverService.java

```diff
@@ -0,0 +1,140 @@
+package com.owncloud.android.services.observer;
+
+import java.io.File;
+import java.util.HashSet;
+
+import com.owncloud.android.MainApp;
+import com.owncloud.android.datamodel.InstantUploadPreference;
+import com.owncloud.android.lib.common.utils.Log_OC;
+
+import android.app.Service;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.os.IBinder;
+import android.widget.Toast;
+
+public class InstantUploadFolderObserverService extends Service {
+    private InstantUploadFolderObserver fileOb;
+    private static final String TAG = "InstantUploadFolderObserverService";
+    //private static final int MAX_FO = 1;
+    //private List<MyFileObserver> fileOb_list = new ArrayList<MyFileObserver>();
+
+    // TODO Tobi: restart Service on change of setting (crud)
+
+    @Override
+    public void onCreate() {
+        // TODO start Observer for each stored Entry
+        File sdcard = new File("/mnt/sdcard/DCIM/Camera/");
```

Comment:
The path to the SD Card should not be hardcoded. Probably we'll need using several methods from https://developer.android.com/reference/android/os/Environment , with different paths depending on Android version.